test/entrypoint/srv_select/storage_init.lua

```diff
@@ -229,6 +230,102 @@ return function()
         if_not_exists = true,
     })
 
+    if decimal_supported then
+        local decimal_format = {
+            {name = 'id', type = 'unsigned'},
+            {name = 'bucket_id', type = 'unsigned'},
+            {name = 'decimal_field', type = 'decimal'},
+        }
+
+
+        local decimal_nonindexed_space = box.schema.space.create('decimal_nonindexed', {
+            if_not_exists = true,
+            engine = engine,
+        })
+
+        decimal_nonindexed_space:format(decimal_format)
+
+        decimal_nonindexed_space:create_index('id_index', {
+            parts = { 'id' },
+            if_not_exists = true,
+        })
+
+        decimal_nonindexed_space:create_index('bucket_id', {
+            parts = { 'bucket_id' },
+            unique = false,
+            if_not_exists = true,
+        })
+
+
+        local decimal_indexed_space = box.schema.space.create('decimal_indexed', {
+            if_not_exists = true,
+            engine = engine,
+        })
+
+        decimal_indexed_space:format(decimal_format)
+
+        decimal_indexed_space:create_index('id_index', {
+            parts = { 'id' },
+            if_not_exists = true,
+        })
+
+        decimal_indexed_space:create_index('bucket_id', {
+            parts = { 'bucket_id' },
+            unique = false,
+            if_not_exists = true,
+        })
+
+        decimal_indexed_space:create_index('decimal_index', {
+            parts = { 'decimal_field' },
+            unique = false,
+            if_not_exists = true,
+        })
+
+
```

Comment:
It was intentional since there is a hundred lines of code and it would be nice to visually separate one space setup from the other. But I can remove them, if you insist.